controllers/kubeeye/clusterinsight_controller.go

```diff
@@ -123,6 +124,28 @@ func (r *ClusterInsightReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 	insightName := clusterInsight.ObjectMeta.Name
 
 	if clusterInsight.Status.AuditPercent == 0 && clusterInsight.Status.AuditResults == nil {
```

Comment:
你可能还不了解k8s的list-watch机制。下面这段写的不对。

generally, 你不需要在reconcile func里使用goroutine，去持续更新一个对象。因为这个对象只要一更新，就会调用reconcile，所以你只需要更新一次，等待下一次reconcile就好。

例如，更新 set precent to 5 -> controller收到一个update event，触发reconcile -》 再次计算percent, 得到10， 于是 set percent to 10 -> controller收到一个update event，触发reconcile  -》 。。。。-〉 set percent to 100 -> controller收到一个update event，触发reconcile -> percent 依旧为100， 不更新 -》 没有event，不触发reconcile

list-watch机制可参考 https://developer.aliyun.com/article/798703

Reply:
from https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.11.0/pkg/reconcile:
```
	// Reconcile performs a full reconciliation for the object referred to by the Request.
	// The Controller will requeue the Request to be processed again if an error is non-nil or
	// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
```